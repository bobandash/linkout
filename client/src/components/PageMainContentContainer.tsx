import { FC } from 'react';

interface PageTemplateProps {
  children: React.ReactNode;
}

const PageMainContentContainer: FC<PageTemplateProps> = ({ children }) => {
  return (
    <div className="flex flex-grow flex-col overflow-y-scroll bg-color_3 p-5 md:p-7 lg:p-10 lg:pt-0 2xl:p-16 2xl:pt-0">
      {children}
    </div>
  );
};

export default PageMainContentContainer;
