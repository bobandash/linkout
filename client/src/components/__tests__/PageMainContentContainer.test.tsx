import { render, screen } from '@testing-library/react';
import PageMainContentContainer from '../PageMainContentContainer';

const MockChild = () => {
  return (
    <div>
      <h1>Test</h1>
    </div>
  );
};

describe('PageMainContentContainer', () => {
  it('renders correctly', () => {
    const { container } = render(
      <PageMainContentContainer>
        <MockChild />
      </PageMainContentContainer>,
    );
    expect(screen.getByText('Test')).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
