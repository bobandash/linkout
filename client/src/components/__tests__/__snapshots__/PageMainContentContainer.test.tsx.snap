// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PageMainContentContainer > renders correctly 1`] = `
<div>
  <div
    class="flex flex-grow flex-col overflow-y-scroll bg-color_3 p-5 scrollbar-thin md:p-7 lg:p-10 lg:pt-0 2xl:p-16 2xl:pt-0"
  >
    <div>
      <h1>
        Test
      </h1>
    </div>
  </div>
</div>
`;
